integration-cli/docker_cli_push_test.go

```diff
@@ -577,4 +577,6 @@ func (s *DockerRegistryAuthTokenSuite) TestPushMisconfiguredTokenServiceResponse
 	dockerCmd(c, "tag", "busybox", repoName)
 	out, _, err := dockerCmdWithError("push", repoName)
 	c.Assert(err, check.NotNil, check.Commentf(out))
+	split := strings.Split(out, "\n")
+	c.Assert(split[len(split)-2], check.Equals, "unauthorized: authentication required")
```

Comment:
@runcom: Makes sense - looks like the special case wouldn't be hit with the old code. https://github.com/docker/distribution/pull/1531/files#diff-de05143a5d591b472e57dd8c46f3998fR88